src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -677,6 +686,7 @@ public void run() {
                     LOG.debug("Error closing server socket", ie);
                 }
             }
+            if (ce !=null) throw new RuntimeException(ce);
```

Comment:
Would it be possible to refactor this so the exception never leaves its catch block? Perhaps the logic "in between" can be captured with "finally". Then I think we will fix https://github.com/apache/zookeeper/pull/257/files#r117407300 as well.